### PR TITLE
Fix problem with transform box of multiscale image 

### LINKED ISSUE #6382: Transform mode problem when zoomed out on multiscale layer
### 🐛 Bug Report

The transform box adopts a wrong shape when zooming out on a multiscale layer.

### 💡 Steps to Reproduce

```python
import napari
from skimage import data
from skimage.transform import pyramids

im = data.astronaut()[:,:,0].astype(float)

viewer = napari.Viewer()
layers = viewer.add_image([im] + [pyramids.pyramid_reduce(im, downscale=s) for s in [2]])

viewer.layers.selection.active.mode = 'transform'

viewer.camera.zoom = 0.1
```

Screenshot:

image

### 💡 Expected Behavior

_No response_

### 🌎 Environment

napari: 0.5.0a2.dev413+gae384aec
Platform: macOS-12.6-arm64-arm-64bit
System: MacOS 12.6
Python: 3.10.12 | packaged by conda-forge | (main, Jun 23 2023, 22:41:52) [Clang 15.0.7 ]
Qt: 5.15.8
PyQt5: 5.15.9
NumPy: 1.26.1
SciPy: 1.11.3
Dask: 2023.10.0
VisPy: 0.14.1
magicgui: 0.8.0
superqt: 0.6.1
in-n-out: 0.1.9
app-model: 0.2.2
npe2: 0.7.3

OpenGL:
  - GL version:  2.1 Metal - 76.3
  - MAX_TEXTURE_SIZE: 16384

Screens:
  - screen 1: resolution 1440x900, scale 2.0
  - screen 2: resolution 1920x1080, scale 2.0
  - screen 3: resolution 1920x1080, scale 2.0

Settings path:
  - /Users/malbert/Library/Application Support/napari/napari_main_613533f96f8fd75b0ca19a87437ef7196a25cf0a/settings.yaml
Plugins:
  - napari: 0.5.0a2.dev413+gae384aec (77 contributions)
  - napari-console: 0.0.9 (0 contributions)
  - napari-manual-transforms: 0.0.3 (2 contributions)
  - napari-svg: 0.1.10 (2 contributions)
  - napari-threedee: 0.0.12 (26 contributions)

### 💡 Additional Context

_No response_

### PR DESCRIPTION
**Pull request by Czaki**
# References and relevant issues

fix #6382

# Description

The problem described in #6392 comes from using `extent_data` for level 0, but the Transform box visual is children of the image visual. When we change the level of data then we update the transform matrix of the image visual which leads to increase size of Transform Box. 

In this PR I add an extent property that reflects the current data level. So We get a smaller extent, which is fixed by scaling on the image layer.

### CODE REVIEW

**Comment by psobolewskiPhD** [comment]
This looks good and I was going to approve, but alas it only fixes the visual of the overlay box when zooming out to trigger the lower scale.
The shown vertexes are not properly intractable, you need to interact with where the vertex would be before this fix. see:

https://github.com/napari/napari/assets/76622105/ead014af-cc60-467c-be5a-39ba3e0a86e3

**Comment by Czaki** [comment]
Hm. When debugging another multiscale error, I lost part of the solution during multiple rebases. Restored it now.

File: napari/layers/image/image.py [hunk 1]
```diff
@@ -475,6 +475,22 @@ def _extent_data_augmented(self) -> np.ndarray:
         extent = self._extent_data
         return extent + [[-0.5], [+0.5]]
 
+    @property
+    def _extent_level_data(self) -> np.ndarray:
+        """Extent of layer in data coordinates.
```
Comment (psobolewskiPhD) [inline napari/layers/image/image.py]:
```suggestion
        """Extent of layer, accounting for current multiscale level, in data coordinates.
```

File: napari/layers/base/base.py [hunk 2]
```diff
@@ -1630,11 +1631,20 @@ def _display_bounding_box_augmented(
...
 
+    def _display_bounding_box_augmented_data_level(
+        self, dims_displayed: List[int]
+    ) -> npt.NDArray:
+        """An augmented, axis-aligned (ndisplay, 2) bounding box.
+
+        If the layer is multiscale layer, then returns the
+        bounding box of the data at the current level
+        """
+        return self._display_bounding_box_augmented(dims_displayed)
```
Comment (psobolewskiPhD) [inline napari/layers/base/base.py:1646]: I don't get how this is accounting for data level. Or is the doc string copy paste?

Comment (Czaki) [reply napari/layers/base/base.py:1646]: Currently, only Image and Labels support multiscale data. So for all other layer types, the implementation will be identical to `_display_bounding_box_augmented_data`. But it is overwritten in `_ImageBase` to provide proper implementation for multiscale data.

**Review by psobolewskiPhD — APPROVED** [review]

**Review by brisvag — APPROVED** [review]

**Comment by m-albert** [comment]
Thank you for fixing this 🎉

**Pull request merged**